azure-devops/global/abstracts/run-test-e2e.yaml

```diff
@@ -16,32 +16,35 @@ steps:
 
   - template: ./replace-environment-variables.yaml
     parameters:
-      FILE: '$(Agent.TempDirectory)/OpenSearch-Dashboards/plugins/${{ parameters.PLUGIN_NAME }}/.ci/config/opensearch_dashboards.yml'
-      DISPLAY_NAME: 'Replace Environment Variables'
+      FILE: "$(Agent.TempDirectory)/OpenSearch-Dashboards/plugins/${{ parameters.PLUGIN_NAME }}/.ci/config/opensearch_dashboards.yml"
+      DISPLAY_NAME: "Replace Environment Variables"
 
   - script: |
       idToken=$(curl -s --request POST --header 'Content-Length: 0' \
           --url "$(ROPC_URL)?username=$(ROPC_USERNAME)&password=$(ROPC_PASSWORD)&grant_type=password&scope=openid%20$(ROPC_CLIENT_ID)%20offline_access&client_id=$(ROPC_CLIENT_ID)&response_type=token%20id_token" | grep -oP '(?<="id_token":")[^"]*')
 
       sed "s/ID_TOKEN/$idToken/" -i $(Agent.TempDirectory)/OpenSearch-Dashboards/plugins/${{ parameters.PLUGIN_NAME }}/public/global/functions/fetch_token.ts
     name: GetCatalogToken
-    displayName: 'Get API Token'
+    displayName: "Get API Token"
 
   - script: |
       cd $(Agent.TempDirectory)/OpenSearch-Dashboards/plugins/${{ parameters.PLUGIN_NAME }}
+      sed -i "s|schema.string()|schema.string({defaultValue: '$(CATALOG_API_URL)'})|" $(Agent.TempDirectory)/OpenSearch-Dashboards/plugins/${{ parameters.PLUGIN_NAME }}/server/index.ts
```

Comment:
Specific information. Please remove it. Find a way of allowing generic passing information.